dbt/include/clickhouse/macros/materializations/incremental.sql

```diff
@@ -234,3 +244,36 @@
     {% do adapter.drop_relation(new_data_relation) %}
     {{ drop_relation_if_exists(distributed_new_data_relation) }}
 {% endmacro %}
+
+{% macro clickhouse__incremental_insert_replace(existing_relation, intermediate_relation, partition_by) %}
```

Comment:
this strategy should also take care about cluster setup and distributed tables. 
We can insert data through Distributed table and then do replace for local table on each shard. This part will be mostly the same as in `clickhouse__incremental_delete_insert` macro

Reply:
Should we maybe defer the implementation till the next iteration? My immediate requirement was to support the strategy for single-node setups (which we are using currently with my team). We may leave a chance to others implement and test the cluster-specific version. I won't have a quick option to test it, thus I won't be able to confirm it works.

Your suggestion with `system.parts` works completely fine, thanks for that! I plan to cover it with tests and document in the next couple of weeks (having a vacation next week :) ).

Reply:
Not sure about merging without Distributed support, because this is new functionality and mostly experimental, I think this is a question to @genzgd, as he is a maintainer.

But I believe that it should work with cluster from beginning because this is core CH functionality, and most of the production users don't use single server setup

Reply:
We use dbt with ClickHouse without cluster currently, so I have not had an opportunity to test it. I suggest on merging the tested version of my PR now. Once someone needs a clustered version of the strategy, they may contribute it too and test themselves.

If I add the clusters-relates logic, I may not guarantee that it works. However, if the maintainers are ok to rely on their review, I am completely fine with adding a related code snippet.